tests/priceEngine.test.ts

```diff
@@ -0,0 +1,148 @@
+import { describe, it, expect, vi, beforeEach } from 'vitest'
+import { PriceEngine } from 'src/core'
+import { ClientOptions } from 'src/fetcher'
+import type { AppraisalOutputCoverage } from 'src/types'
+
+describe('PriceEngine', () => {
+  let fetchMock: any
+  const clientOptions: ClientOptions = {
+    apiKey: '',
+    baseURL: 'https://api.dd360.mx',
+    timeout: 5000,
+    maxRetries: 2
+  }
+  beforeEach(() => {
+    fetchMock = vi.fn()
+    global.fetch = fetchMock
+  })
+
+  it('should throw an error if API key is missing', () => {
+    expect(() => new PriceEngine({})).toThrow('API Key is required')
+  })
+
+  it('should create an instance of PriceEngine', () => {
+    const fetcher = new PriceEngine(clientOptions)
+    expect(fetcher).toBeInstanceOf(PriceEngine)
+    expect(fetcher.getVersion()).toBe('v9')
+  })
+
+  it('should get appraisal coverage', async () => {
+    const request = {
+      latitude: 19.41712064426177,
+      longitude: -99.17343830677876
+    }
+    const response: AppraisalOutputCoverage = {
+      hasCoverage: true,
+      cvegeo: '0901500011182008'
+    }
+    fetchMock.mockResolvedValue({
+      ok: true,
+      json: () => Promise.resolve({ data: response })
+    })
+    const priceEngine = new PriceEngine(clientOptions)
+    const result = await priceEngine.getAppraisalCoverage(request)
+    expect(result).toEqual(response)
+  })
+
+  it('should handle API errors correctly on wrong input format', async () => {
+    // Wrong format for latitude and longitude
+    const request = {
+      latitudi: '19.41712064426177',
+      longitudo: '-99.17343830677876'
+    }
+    fetchMock.mockResolvedValue({
+      ok: false,
+      status: 400,
+      statusText: 'Bad Request'
+    })
+    const priceEngine = new PriceEngine(clientOptions)
+    await expect(
+      priceEngine.getAppraisalCoverage(request as any)
+    ).rejects.toThrow('Bad Request: Bad Request')
+  })
+
+  it('should get appraisal', async () => {
+    const request = {
+      latitude: 19.41712064426177,
+      longitude: -99.17343830677876,
+      lotSurface: 644,
+      constructionArea: 702,
+      propertyType: 'house',
+      appraisalType: 'sale'
+    }
+    const response = {
+      value: 29869257.6,
+      valuePerSqm: 40600,
+      upperValueRangePerSqm: 45472.00000000001,
+      lowerValueRangePerSqm: 35728,
+      upperValueRange: 31921344.000000004,
+      lowerValueRange: 25081056
+    }
+    fetchMock.mockResolvedValue({
+      ok: true,
+      json: () => Promise.resolve({ data: response })
+    })
+
+    const priceEngine = new PriceEngine(clientOptions)
+    const result = await priceEngine.getAppraisal(request)
+    expect(result).contain(response)
+  })
+
+  it('should get appraisal for apartment rent', async () => {
+    const request = {
+      latitude: 19.41712064426177,
+      longitude: -99.17343830677876,
+      lotSurface: 644,
+      constructionArea: 702,
+      propertyType: 'house',
```

Comment:
```suggestion
```

The payload need changes! If you are testing the variables methods like `getApartmentRent` => You don't need pass the propertyType and appraisalType